executor/show.go

```diff
@@ -1876,16 +1877,18 @@ func (e *ShowExec) fetchShowWarnings(errOnly bool) error {
 
 // fetchShowPumpOrDrainerStatus gets status of all pumps or drainers and fill them into e.rows.
 func (e *ShowExec) fetchShowPumpOrDrainerStatus(kind string) error {
-	registry, err := createRegistry(config.GetGlobalConfig().Path)
+	registry, needToClose, err := createRegistry(config.GetGlobalConfig().Path)
```

Comment:
Is it better to use a new function? This client, I think, is used for Pump/Drainer specifically.

Reply:
How about changing function's name to avoid abuse?